Lib/zoneinfo/_common.py

```diff
@@ -22,6 +22,23 @@ def load_tzdata(key):
         # UnicodeEncodeError: If package_name or resource_name are not UTF-8,
         #   such as keys containing a surrogate character.
         raise ZoneInfoNotFoundError(f"No time zone found with key {key}")
+    except OSError as exc:
+        # Exceptions inherited from OSError can be raised in various scenarios:
+        #
+        # IsADirectoryError: If the resource_name links to a directory
+        #   (e.g. Australia)
+        # PermissionError: If the resource_name links to a directory on Windows
+        #   (e.g. Pacific)
+        import os
+        import importlib.util
+
+        package_spec = importlib.util.find_spec(package_name)
+        resource_path = os.path.join(os.path.dirname(package_spec.origin), key)
```

Comment:
imports+two lines of code is a risk of raising a new exception. You may do that before and use the path to load the resource. I'm not sure.

Is it possible that tzdata is in a ZIP file, or in memory but not on disk?

Maybe it's overcomplicated and maybe it's better to convert *any* OSError into a ZoneInfoNotFoundError. I let @pganssle decides what is the best.

Reply:
Yeah, I don't like this at all. In `importlib_resources` there's a way to determine if something is a directory or not using the [`importlib.resources.files`](https://docs.python.org/3.9/library/importlib.html#importlib.resources.files) API.

That will complicate things a bit for the backport, though, which doesn't currently require `importlib_resources` in Python 3.8.

I'm inclined to go the simple way and catch *all* `OSError`s instead of the more narrowly scoped ones. I realize this may frustrate some people debugging in unusual environments by disguising the source of some errors, but I don't love that the standard `ZoneInfo` constructor is exposing details about the specifics of the underlying file system.

Reply:
@jaraco Do you know if there's an equivalent of `importlib.resources.files('my.package').joinpath('resource').isdir()` that works for Python 3.8's version of `importlib.resources`?

Reply:
In Python 3.8 and earlier, `importlib.resources` didn't support directories as resources - all resources were required to be files directly in a package.

Reply:
@jaraco Well, we're not really concerned with using a directory as a resource, the problem is that it appears that `importlib.resources` raises `PermissionError` on Windows when a directory is specified, and we're trying to distinguish between the case where a purported resource *exists but is a directory* and the case of a legitimate permissions error.

Is there no way to do that in the Python 3.8 `importlib.resources`?

Reply:
swapped to using `importlib.resources.files` to locate the resource however inclined to agree that catching all OSError derivatives might prove less problematic.